### PR TITLE
Remove transation outcome and fix changes

### PR DESCRIPTION
**Pull request by danielailie**

### CODE REVIEW

File: src/tokenManagement/tokenManagementTransactionsOutcomeParser.ts [hunk 1]
```diff
@@ -441,7 +390,7 @@ export class TokenManagementTransactionsOutcomeParser {
             return "";
         }
         const address = Buffer.from(event.topics[3]);
-        return Address.fromBuffer(address).bech32();
+        return new Address(address).bech32();
```
Comment (popenta) [inline src/tokenManagement/tokenManagementTransactionsOutcomeParser.ts:393]: use `toBech32` here, as well.

File: src/transactionEvents.ts [hunk 2]
```diff
@@ -0,0 +1,41 @@
+import { Address } from "./address";
+
+export class TransactionEvent {
+    address: Address = Address.empty();
+    identifier: string = "";
+    topics: Uint8Array[] = [];
+
+    data: Uint8Array = new Uint8Array();
+    additionalData: Uint8Array[] = [];
```
Comment (andreibancioiu) [inline src/transactionEvents.ts:9]: :+1:

File: src/transactionEvents.ts [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+
+    constructor(init?: Partial<TransactionEvent>) {
+        Object.assign(this, init);
+    }
+
+    static fromHttpResponse(responsePart: {
```
Comment (andreibancioiu) [inline src/transactionEvents.ts:15]: In sdk-py, this conversion is defined somewhere else (network providers module).

File: src/transactionEvents.ts [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+        address: string;
+        identifier: string;
+        topics: string[];
+        data: string;
+        additionalData?: string[];
+    }): TransactionEvent {
+        let result = new TransactionEvent();
+        result.address = new Address(responsePart.address);
+        result.identifier = responsePart.identifier || "";
+        result.topics = (responsePart.topics || []).map((topic) => Buffer.from(topic));
```
Comment (andreibancioiu) [inline src/transactionEvents.ts:25]: Decode as base64, instead?

https://github.com/multiversx/mx-sdk-py/blob/feat/next/multiversx_sdk/network_providers/http_resources.py#L186

File: src/transactionEvents.ts [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+        return result;
+    }
+
+    findFirstOrNoneTopic(predicate: (topic: Uint8Array) => boolean): Uint8Array | undefined {
+        return this.topics.filter((topic) => predicate(topic))[0];
+    }
+
+    getLastTopic(): Uint8Array {
+        return this.topics[this.topics.length - 1];
+    }
```
Comment (andreibancioiu) [inline src/transactionEvents.ts:40]: If they aren't used anymore, can be dropped (sorry if I'm mistaken, I didn't double check).

File: src/networkProviders/contractResults.ts
[the remark this answers is not in the document]
Comment (danielailie) [reply src/networkProviders/contractResults.ts:1]: I will remove them in the next pr when I remove the result parsers

File: src/transaction.ts [hunk 6]
```diff
@@ -178,21 +177,21 @@ export class Transaction {
     /**
      * Legacy method, use the "sender" property instead.
      */
-    getSender(): IAddress {
+    getSender(): Address {
```
[the remark this answers is not in the document]
Comment (danielailie) [reply src/transaction.ts:180]: done

**Review by andreibancioiu — APPROVED** [review]

**Review by popenta — APPROVED** [review]

**Pull request merged**